st2common/st2common/models/db/stormbase.py

```diff
@@ -105,6 +107,26 @@ def to_python(self, value):
         return mongoescape.unescape_chars(value)
 
 
+class TagField(me.EmbeddedDocument):
+    """
+    To be attached to a db model object for the purpose of providing supplemental
+    information.
+    """
+    name = me.StringField()
```

Comment:
Yes this approach helps with indexing. I have not fully thought through the storage implications. The cost of storing tags and the size of the index itself could become a problem. As a mitigation step I have decided to limit tags to Action, Rules and TriggerTypes which should for the near term be in 100s or 1000s. Staying away from tagging ActionExecutions and TriggerTypes as those are already in 10^5 or 10^6.